controllers/cps_controller_helper.go

```diff
@@ -0,0 +1,217 @@
+		err = r.addConstraintStatusManager(cpsMgr)
+		if err != nil {
+			r.isCpsCtlInstalled = false
+
+			setupLog.Error(err, "unable to start ConstraintPodStatus manager")
+		}
+	}()
+
+	return nil
+}
+
+func (r *GatekeeperReconciler) addConstraintStatusManager(mgr manager.Manager,
+) error {
+	if err := (&ConstraintStatusReconciler{
+		Scheme:               r.Scheme,
+		Client:               mgr.GetClient(),
+		ClientSet:            kubernetes.NewForConfigOrDie(mgr.GetConfig()),
+		DynamicClient:        r.DynamicClient,
+		Log:                  ctrl.Log.WithName("controllers").WithName("ConstraintPodStatus"),
+		Namespace:            r.Namespace,
+		ConstraintToSyncOnly: map[string][]gkv1alpha1.SyncOnlyEntry{},
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ConstraintPodStatus")
+
+		return err
+	}
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+
+		return err
+	}
+
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+
+		return err
+	}
+
+	r.isCpsCtlInstalled = true
+
+	if err := mgr.Start(r.childCtx); err != nil {
+		setupLog.Error(err, "problem running ConstraintStatus manager")
+
+		return err
+	}
+
+	return nil
+}
+
+// Check constraintPodStatus crd status is "True" and type is "NamesAccepted"
+func checkCPSCrdAvailable(mainCtx context.Context, dynamicClient *dynamic.DynamicClient) bool {
+	crdGVR := schema.GroupVersionResource{
+		Group:    "apiextensions.k8s.io",
+		Version:  "v1",
+		Resource: "customresourcedefinitions",
+	}
+
+	crd, err := dynamicClient.Resource(crdGVR).
+		Get(mainCtx, "constraintpodstatuses.status.gatekeeper.sh", metav1.GetOptions{})
+	if err != nil {
+		setupLog.V(1).Info("Cannot fetch ConstraintPodStatus crd")
+
+		return false
+	}
+
+	conditions, ok, _ := unstructured.NestedSlice(crd.Object, "status", "conditions")
+	if !ok {
+		setupLog.V(1).Info("Cannot parse ConstraintPodStatus status conditions")
+
+		return false
+	}
+
+	for _, condition := range conditions {
+		parsedCondition := condition.(map[string]interface{})
+
+		status, ok := parsedCondition["status"].(string)
+		if !ok {
+			setupLog.V(1).Info("Cannot parse ConstraintPodStatus conditions status")
+
+			return false
+		}
+
+		conditionType, ok := parsedCondition["type"].(string)
+		if !ok {
+			setupLog.V(1).Info("Cannot parse ConstraintPodStatus conditions type")
+
+			return false
+		}
+
+		if conditionType == "NamesAccepted" && status == "True" {
+			setupLog.V(1).Info("ConstraintPodStatus crd is ready")
+
+			return true
+		}
+	}
+
+	setupLog.V(1).Info("ConstraintPodStatus crd is not ready yet")
+
+	return false
+}
+
+// Check gatekeeper auditFromCache=Automatic
+// Check constraintpodstatuses.status.gatekeeper.sh crd status has "NamesAccepted" condition that is true.
+func checkCPScontrollerAvailable(mainCtx context.Context,
```

Comment:
This is a better description of what it does.
```suggestion
func checkCPScontrollerPrereqs(mainCtx context.Context,
```